src/inversion_scripts/operators/operator_utilities.py

```diff
@@ -52,15 +80,95 @@ def read_geoschem(date, gc_cache, build_jacobian=False, sensi_cache=None):
+        # perturbation simulation numbers
+        pert_simulations_dict = {}
+        for e in elements:
+            # State vector elements are numbered 1..nelements
+            sv_elem = e + 1
+        
+            is_OH_element = check_is_OH_element(sv_elem, n_elements, opt_OH)
+            # Determine which run directory to look in
+            if is_OH_element:
+                run_number = nruns
+            elif check_is_BC_element(sv_elem, n_elements, opt_OH, opt_BC, is_OH_element):
+                num_back = n_elements % sv_elem
+                run_number = nruns - num_back
+            else:
+                run_number = np.ceil(sv_elem / ntracers).astype(int)
+        
+            run_num = str(run_number).zfill(4)
+        
+            # add the element to the dictionary for the relevant simulation number
+            if run_num not in pert_simulations_dict:
+                pert_simulations_dict[run_num] = [sv_elem]
+            else:
+                pert_simulations_dict[run_num].append(sv_elem)
+                
+                
+        gc_date = pd.to_datetime(date, format='%Y%m%d_%H')
+        ds_all = [concat_tracers(k, gc_date, config, v) for k,v in pert_simulations_dict.items()]
+        ds_sensi = xr.concat(ds_all, 'element')
+        ds_sensi.load()
+        
+        sensitivities = ds_sensi["ch4"].values
         # Reshape so the data have dimensions (lon, lat, lev, grid_element)
         sensitivities = np.einsum("klji->ijlk", sensitivities)
-        sensitivity_data.close()
-        dat["Sensitivities"] = sensitivities
+        dat["jacobian_ch4"] = sensitivities
+        
+        # get emis base, which is also BC base
+        ds_emis_base = concat_tracers('0001', gc_date, config, [0], baserun=True)
+        ds_emis_base.load()
+        dat['emis_base_ch4'] = np.einsum('klji->ijlk', ds_emis_base['ch4'].values)
+        
+        # get OH base, run RunName_0000
+        # it's always here whether OptimizeOH is true or not
+        # so we can keep it here for convenience
+        ds_oh_base = concat_tracers('0000', gc_date, config, [0])
+        ds_oh_base.load()
+        dat['oh_base_ch4'] = np.einsum('klji->ijlk', ds_oh_base['ch4'].values)
+        
 
     return dat
 
+
+def concat_tracers(run_id, gc_date, config, sv_elems, baserun=False):
+    prefix = config['OutputPath'] + '/' + config['RunName'] + '/jacobian_runs'
+    j_dir = f"{prefix}/{config['RunName']}_{run_id}/OutputDir"
+    file_stub = gc_date.strftime('GEOSChem.SpeciesConc.%Y%m%d_0000z.nc4')
+    dsmf = xr.open_dataset(
+        '/'.join([j_dir,file_stub]),
+        chunks = 'auto'
+    )
+    keepvars = [f'SpeciesConcVV_CH4_{i:04}' for i in sv_elems]
+    if len(keepvars) == 1:
+        # for BC and OH elems, no number in var name
+        keepvars = ['SpeciesConcVV_CH4']
+    elif baserun:
+        keepvars = ['SpeciesConcVV_CH4']
+    ds_concat = xr.concat([dsmf[v] for v in keepvars], 'element').rename('ch4')
```

Comment:
Shouldnt non-baserun elements have a keepvar of `SpeciesConcVV_CH4_000X`. I think I am misunderstanding something here?

Reply:
Expanded the doc string for a better explanation. In each jacobian simulation, there's a variable `SpeciesConcVV_CH4` with no number appended, if this switch is true then it grabs this variable instead of the tracer variables. It is used for the base in the sensitivity calculation for emissions perturbation elements.